pkg/tfgen/generate_nodejs.go

```diff
@@ -479,6 +479,29 @@ func (g *nodeJSGenerator) emitResourceType(mod *module, res *resourceType) (stri
 		w.Writefmtln("")
 	}
 
+	w.Writefmtln("    private static readonly __pulumiType = '%s';", res.info.Tok)
+
+	if !res.IsProvider() {
```

Comment:
Why do we leave this out for providers?  It seems reasonable to want to write queries that use `isInstance` that look at provider instances and verify things about them.